### PR TITLE
Manifest version 2

### PR DESCRIPTION
**Pull request by bjoerge**
This changes the template manifest version to cater for Vercel as deployment provider.

*Notable changes*:

 - Bumping manifest spec version from `v0`/`v1` to `v2`. Since we have had no strict validation of version up until now there's already several templates out there with `version: 1`, so these changes becomes v2 instead. As an added bonus the manifest version now also match the npm package version.
- Defined the template manifest typings using the excellent [zod](https://github.com/vriad/zod)-package. Which means we can import the manifest type and get type safe templates in applications that reads them.
- Added a cli command for migrating from v0/v1 to v2 by running `npx sanity-template migrate` from the template repository root folder.
- Added typescript to the project itself (although there's still a few js files left)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .DS_Store
 /node_modules
+dist
```
Comment (mariuslundgard) [inline .gitignore]: Should be `/dist` to not ignore sub-directories

File: SPEC-MANIFEST-V1.md [hunk 2]
```diff
@@ -98,10 +104,13 @@ interface DeploymentDeclaration {
     };
     requirements?: NetlifyRequirement[];
   };
+]
```
Comment (mariuslundgard) [inline SPEC-MANIFEST-V1.md]: There's some indentation missing here, which makes it harder to read.

File: src/commands/generateLockFiles.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 
-const {temporaryTransformPackage} = require('./lib/temporaryTransformPackage')
-const {runNpmInstall} = require('./lib/runNpmInstall')
+const {temporaryTransformPackage} = require('./utils/temporaryTransformPackage')
+const {runNpmInstall} = require('./utils/runNpmInstall')
 
 const path = require('path')
 
-function generateLockFiles({basedir}) {
+function generateLockFiles(basedir) {
```
Comment (mariuslundgard) [inline src/commands/generateLockFiles.js]: Changing from `{basedir}` to `basedir` makes the CLI command `sanity-template lockfiles` fail:

```
The "path" argument must be of type string. Received an instance of Object
    at validateString (internal/validators.js:117:11)
    at Object.join (path.js:1039:7)
```

**Comment by bjoerge** [comment]
Merged

**Pull request closed without merging**